sdk/eventgrid/Azure.Messaging.EventGrid/EventGridSourceGenerator/src/EventGridSourceGenerator.cs

```diff
@@ -13,38 +18,168 @@ namespace Azure.EventGrid.Messaging.SourceGeneration
+            var eventTypeSet = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var classDeclaration in classes)
+            {
+                var semanticModel = compilation.GetSemanticModel(classDeclaration.SyntaxTree);
+                if (semanticModel.GetDeclaredSymbol(classDeclaration) is not INamedTypeSymbol classSymbol)
+                {
+                    continue;
+                }
+
+                var documentationCommentXml = classSymbol.GetDocumentationCommentXml();
+                if (string.IsNullOrEmpty(documentationCommentXml))
+                {
+                    continue;
+                }
+
+                // Extract event type from documentation comments
+                string eventType = ExtractEventTypeFromDocumentation(documentationCommentXml);
+                if (string.IsNullOrEmpty(eventType))
+                {
+                    // Skip if no event type is found (likely a base type)
+                    continue;
+                }
+
+                if (!eventTypeSet.Add(eventType))
+                {
+                    continue;
+                }
+
+                // Find the deserialize method
+                var deserializeMethod = classSymbol.GetMembers()
+                    .OfType<IMethodSymbol>()
+                    .FirstOrDefault(m => m.Name.StartsWith("Deserialize", StringComparison.Ordinal))?.Name;
+
+                if (deserializeMethod == null)
+                {
+                    // Skip if no deserialize method is found
+                    continue;
+                }
+
+                // Create a SystemEventNode for this event
+                systemEventNodes.Add(new SystemEventNode(eventName: classSymbol.Name, eventType: $@"""{eventType}""", deserializeMethod: deserializeMethod));
+            }
+
+            return systemEventNodes;
+        }
+
+        private static string ExtractEventTypeFromDocumentation(string documentationCommentXml)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(documentationCommentXml))
+                {
+                    return null;
+                }
+
+                ReadOnlySpan<char> docSpan = documentationCommentXml.AsSpan();
+
+                int summaryStartIndex = docSpan.IndexOf(SummaryStartTag);
+                if (summaryStartIndex < 0)
+                {
+                    return null;
+                }
+
+                summaryStartIndex += SummaryStartTag.Length;
+
+                int summaryEndIndex = docSpan.Slice(summaryStartIndex).IndexOf(SummaryEndTag);
+                if (summaryEndIndex < 0)
+                {
+                    return null;
+                }
+
+                var summaryContent = docSpan.Slice(summaryStartIndex, summaryEndIndex);
+
+                var match = EventTypeRegex.Match(summaryContent.ToString());
+                return match.Success ? match.Value : null;
+            }
+            catch
```

Comment:
Do we need this catch all?  The conditional checks along the way should ensure we don't hit any exceptions, and given the PR mentioned a perf motivator the try/catch adds a perf hit.

Can we add some unit tests to this method to ensure it doesn't blow up in edge cases?

Reply:
Na I don't think so. The previous code had no catch and I was overly defensive. I'll remove it

Reply:
I checked it in can blow up when the input is null or the regex timed out. The input being null should not happen and the timeout wasn't assumed to happen in the previous version either.